src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -832,13 +832,196 @@ theorem contains_keys [EquivBEq α] [LawfulHashable α] (h : m.1.WF) {k : α} :
+  suffices ∀ (t : { m' // ∀ (P : Raw₀ α β → Prop),
+    (∀ {m'' : Raw₀ α β} {a : α} {b : β a}, P m'' → P (m''.insert a b)) → P m → P m' }),
+      (List.foldl (fun m' p => ⟨m'.val.insert p.1 p.2, fun P h₁ h₂ => h₁ (m'.2 _ h₁ h₂)⟩) t l).val =
+    List.foldl (fun m' p => m'.insert p.fst p.snd) t.val l from this _
+  intro t
+  induction l generalizing m with
+  | nil => simp
+  | cons h t ih =>
+    simp
+    rw [ih]
+
+@[simp]
+theorem insertMany_eq_insertList (m : Raw₀ α β) (l : List ((a : α) × β a)) :
+    (insertMany m l).val = insertList m l := by
+  rw [insertList_eq_foldl, insertMany_val]
+
+@[simp]
+theorem insertList_nil:
+    m.insertList [] = m := by
+  simp[insertList, Id.run]
+
+@[simp]
+theorem insertList_singleton [EquivBEq α] [LawfulHashable α] {k: α} {v: β k}:
+    m.insertList [⟨k,v⟩] = m.insert k v := by
+  simp[insertList, Id.run]
+
+@[simp]
+theorem insertList_cons {l: List ((a:α) × (β a))} {k: α} {v: β k}:
+    m.insertList (⟨k,v⟩::l) = (m.insert k v).insertList l := by
+  cases l with
+  | nil => simp[insertList]
+  | cons hd tl => simp[insertList]
+
+theorem insertList_insertList {l1 l2: List ((a:α) × (β a))}:
+    (m.insertList l1).insertList l2 = m.insertList (l1++l2) := by
+  simp[insertList_eq_insertListₘ]
+  induction l1 generalizing m with
+  | nil => simp[insertListₘ]
+  | cons hd tl ih =>
+    simp[insertListₘ]
+    apply ih
+
+theorem insertList_insert {l: List ((a:α) × (β a))} {k: α} {v: β k}:
+    (m.insertList l).insert k v = m.insertList (l ++ [⟨k,v⟩]) := by
+  simp [insertList_eq_insertListₘ]
+  induction l generalizing m with
+  | nil => simp[insertListₘ]
+  | cons hd tl ih =>
+    simp[insertListₘ]
+    rw [ih]
+
+theorem contains_insertList [EquivBEq α] [LawfulHashable α] (h : m.1.WF) {l: List ((a:α) × (β a))} {k: α}:
+    (m.insertList l).contains k ↔ m.contains k ∨ (l.map Sigma.fst).contains k := by
+  simp_to_model using List.containsKey_insertList
+
+theorem contains_insertList_of_mem_list [LawfulBEq α] [LawfulHashable α] (h : m.1.WF) {l: List ((a:α) × (β a))} {k k': α} (k_eq: k == k') {v: β k}:
+    ⟨k,v⟩ ∈ l → (m.insertList l).contains k' := by
+  simp_to_model using contains_insertList_of_mem
+
+
+theorem contains_of_contains_insertList [EquivBEq α] [LawfulHashable α] (h : m.1.WF) {l: List ((a:α) × (β a))} {k: α} :
+    (m.insertList l).contains k → (l.map Sigma.fst).contains k = false → m.contains k := by
+  simp_to_model using List.containsKey_of_containsKey_insertList
+
+theorem contains_insertList_of_contains_map [LawfulBEq α] [LawfulHashable α] (h : m.1.WF) {l: List ((a:α) × (β a))} {k k': α} (k_eq: k == k'):
+    m.contains k = true → (m.insertList l).contains k' := by
+  simp_to_model using contains_insertList_of_contains_first
+
+theorem get?_insertList_mem_list [LawfulBEq α] [LawfulHashable α] {l: List ((a:α) × (β a))} {k k': α} {v: β k} {distinct: List.Pairwise (fun a b => ! a.1 == b.1) l} {k_eq: k == k'} {mem: ⟨k,v⟩ ∈ l} (h: m.1.WF):
```

Comment:
`LawfulBEq` implies `LawfulHashable`, so no need to assume it again here.

Reply:
This has been addressed as far as I can see.